File: 코딩 기초 트레이닝/배열 만들기6-p81859/배열 만들기6.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181859


function solution(arr) {
    const answer = [];
    let i = 0;

    while(i < arr.length) {
        if(answer.length === 0) {
            answer.push(arr[i]);
            i += 1;
        } else if(answer.length > 0 && answer[answer.length -1] === arr[i]) {
            answer.pop();
            i += 1;
        } else if(answer.length > 0 && answer[answer.length -1] !== arr[i]) {
            answer.push(arr[i]);
            i += 1;
        }
    }

    return answer.length > 0 ? answer : [-1];
}

File: 코딩 기초 트레이닝/배열 만들기6-p81859/배열 만들기6.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181859

def solution(arr):
    answer = []
    i = 0

    while i < len(arr) :
        if len(answer) == 0:
            answer.append(arr[i])
            i += 1
        elif len(answer) > 0 and answer[len(answer)-1] == arr[i]:
            del answer[len(answer)-1]
            i += 1
        elif len(answer) > 0 and answer[len(answer)-1] != arr[i]:
            answer.append(arr[i])
            i += 1

    return [-1] if len(answer) == 0 else answer
